source/extensions/filters/network/redis_proxy/command_splitter_impl.h

```diff
@@ -188,7 +189,7 @@ class FragmentedRequest : public SplitRequestBase {
 
   SplitCallbacks& callbacks_;
 
-  Common::Redis::RespValuePtr incoming_request_;
+  Common::Redis::RespValueSharedPtr incoming_request_;
```

Comment:
Sorry why is this a shared pointer again? Where is it being held? Both inside the splitter and in the router? Why is this the case again? Can you add more comments?

Reply:
This a shared pointer because the composite arrays are holding a shared pointer to the base arrays for mget/mset... etc.  

If it's only for this change, the composite array don't need the shared pointer.  But in the next PR, this incoming_request_ will be removed, the connection pool will be holding on to either the incoming_request_ or the composite array that hold on to the shared_pointer.